src/authenticator.rs

```diff
@@ -326,7 +344,15 @@ where
     fn load_credential(&mut self, label: &[u8]) -> Option<Credential> {
         let filename = self.filename_for_label(label);
 
-        let credential: Credential = self.state.try_read_file(&mut self.trussed, filename).ok()?;
+        let mut credential: Credential =
+            self.state.try_read_file(&mut self.trussed, filename).ok()?;
+        // Set the default EncryptionKeyType as PinBased for backwards compatibility
+        // All the new records should have it set as HardwareBased, if not overridden by user
+        credential.encryption_key_type = Some(
+            credential
+                .encryption_key_type
+                .unwrap_or(EncryptionKeyType::default_for_loading_credential()),
+        );
```

Comment:
I’d prefer this to make it clear that we want to override `None`:

```
if credential.encryption_key_type.is_none() {
    credential.encryption_key_type = Some(...);
}
```

Reply:
Ah! I see it now. Indeed that's a  weird construction - it must have been grown organically. Will be corrected.